### DISCUSSION TITLE
colGroup retirement

### DISCUSSION THREAD

**Discussion by MKaramavrov** [primefaces/community#636] [PrimeNG] [answered]
Hello, I've just updated primeNg version from 12 to 16++ and now any width in my table is working. The latest changelog says, that you've retired colgroup template, and I'm trying to figure out how to make my vertical and horizontal scrollable table have back the width. Do you have some template for replacement or there is any other cure for this?

```
<ng-template pTemplate="colgroup" let-columns>
                        <colgroup>
                            <col [style.width]="'38px'">
                            <ng-container *ngFor="let col of columns">
                                <col [style.width]="col.width+50+'px'">
                            </ng-container>
                        </colgroup>
                    </ng-template>

```
I was trying to assign style.width directly to <td> inside my code, but it doesn't apply any width to a cell in a table.
there were some syggestions in stackoverflow e.g. https://stackoverflow.com/questions/67612002/primeng-table-is-not-taking-the-fixed-width-in-angular which nevertheless didn't help.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

You can check the docs PrimeNG still has vertical and horizontal scroll, https://primeng.org/table#scroll

**Reply by MKaramavrov** [reply]
But whenever I mark the table as scrollable, that still doesn't apply the width for columns from colGroup template.
